java/sage/media/format/MPEGParser2.java

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright 2015 The SageTV Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sage.media.format;
+
+import sage.Sage;
+
+import java.io.OutputStream;
+
+public class MPEGParser2
+{
+  public static final int REMUX_TS = 1;
+  public static final int REMUX_PS = 16;
+
+  public enum StreamFormat
+  {
+    /**
+     * The stream doesn't have a specific format. Select the first channel. (This may result in
+     * audio only.)
+     */
+    UNKNOWN,
+    /**
+     * The stream is formatted as ATSC. Searches for the desired streams will be based on this.
+     * (This will always have video.)
+     */
+    ATSC,
+    /**
+     * The stream is formatted as DVB.  Searches for the desired streams will be based on this.
+     */
+    DVB,
+    /**
+     * The stream doesn't have a specific format. (This may result in audio only.)
+     */
+    FREE
+  }
+
+  public enum SubFormat
+  {
+    UNKNOWN,
+    TERRESTRIAL,
+    CABLE,
+    SATELLITE
+  }
+
+  public enum TuneStringType
+  {
+    CHANNEL,
+    PROGRAM,
+    XX_XX,
+    XX_XX_XX,
+    AUTO
+  }
+
+  static
+  {
+    if (System.getProperty("os.name").toLowerCase().contains("windows") ||
+        System.getProperty("os.name").toLowerCase().contains("linux"))
+    {
+      try
+      {
+        System.loadLibrary("JavaRemuxer2");
+      }
+      catch (Throwable t)
+      {
+        System.out.println("ERROR Loading native JavaRemuxer2 library of: " + t);
+      }
+    }
+  }
+
+  public static boolean remuxFile(java.io.File inputFile, java.io.File outputFile, int outputFormat, boolean isTV)
+  {
+    java.io.FileInputStream inputStream = null;
+    java.io.BufferedOutputStream outputStream = null;
+    Remuxer2 remuxer = null;
+
+    try
+    {
+      inputStream = new java.io.FileInputStream(inputFile);
+      outputStream = new java.io.BufferedOutputStream(new java.io.FileOutputStream(outputFile));
+      remuxer = openRemuxer(REMUX_TS, outputFormat, isTV ? StreamFormat.ATSC : StreamFormat.FREE,
+          SubFormat.UNKNOWN, TuneStringType.CHANNEL, 0, 0, 0, 0, 0, outputStream);
+
+      if (remuxer == null)
+        return false;
+
+      byte[] buf = new byte[8084];
```

Comment:
It's a multiple of 188. I can push a fairly large amount of data through to the remuxer at a time, but if it's not 188 byte aligned, then the remuxer provides alignment feedback. To avoid even needing that feedback, I have made the buffers multiples of 188. I'll spell that out better in the code by showing the math.